core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.java

```diff
@@ -141,6 +146,15 @@ protected void onCreate(Bundle savedInstanceState) {
     recyclerView.setAdapter(historyAdapter);
     layoutManager = (LinearLayoutManager) recyclerView.getLayoutManager();
     recyclerView.setLayoutManager(layoutManager);
+    checkEmpty();
```

Comment:
No need to add this to `onCreate` when you're adding it to `onResume`.

Reply:
You could in fact register a data observer on the adapter and perform the check then in `onChanged`, right now the code is brittle and this needs to be called each time the data in the adapter changes

Reply:
@yashk2000 I would like to see the usage of `registerDataObserver` on the adapter as it more closely aligns with the intention of "when the adapter's data changes check if it is empty and set the visibility of the emptyView"

Reply:
@macgills done :rocket: 
Please review again. 
Thanks :)

Reply:
perfect